seqr/views/apis/individual_api_tests.py

```diff
@@ -625,21 +626,79 @@ def test_get_individual_rna_seq_data(self):
+                    'tissueType': None,
+                }],
+                'ENSG00000268903': [{
                     'geneId': 'ENSG00000268903', 'zScore': 7.08, 'pValue':0.000000000588, 'pAdjust': 0.00000000139,
                     'isSignificant': True,
-                },
-            }}
-        })
+                    'tissueType': None,
+                }],
+            },
+            'spliceOutliers': {
+                'ENSG00000106554': [{
+                    'chrom': '7', 'deltaPsi': 0.85, 'end': 132886973, 'geneId': 'ENSG00000106554', 'isSignificant': True,
+                    'pValue': 1.08e-56, 'rareDiseaseSamplesTotal': 20, 'rareDiseaseSamplesWithJunction': 1,
+                    'readCount': 1297, 'start': 132885746, 'strand': '*', 'type': 'psi5', 'zScore': 12.34,
+                    'tissueType': 'F',
+                }]
+            },
+        }})
         self.assertSetEqual(set(response_json['genesById'].keys()), {'ENSG00000135953', 'ENSG00000268903'})
 
+    @mock.patch('seqr.views.apis.individual_api.MAX_SIGNIFICANT_OUTLIER_NUM', 4)
+    def test_get_individual_rna_seq_data_is_significant(self):
+        url = reverse(get_individual_rna_seq_data, args=[INDIVIDUAL_GUID])
+        self.check_collaborator_login(url)
+
+        fibs_sample = Sample.objects.get(individual__guid=INDIVIDUAL_GUID, tissue_type='F')
+        muscle_sample = Sample(
+            individual=Individual.objects.get(guid=INDIVIDUAL_GUID),
+            sample_type='RNA',
+            dataset_type='VARIANTS',
+            sample_id='NA19675_1_1',
+            tissue_type='M',
+            data_source='muscle_samples.tsv.gz',
+            loaded_date=timezone.now(),
+            is_active=True,
+        )
+        muscle_sample.save()
+        test_splice_outliers = [
+            {'start': pos*1000 + 1, 'end': pos*1000 + 100, 'sample': sample, 'p_value': p_value}
+            for pos, (sample, p_value) in enumerate([
+                (fibs_sample, 0.0001), (fibs_sample, 0.001), (fibs_sample, 0.001), (fibs_sample, 0.003), (fibs_sample, 0.1),
+                (muscle_sample, 0.001), (muscle_sample, 0.002), (muscle_sample, 0.2), (muscle_sample, 0.1), (muscle_sample, 0.1),
+            ])
+        ]
+        for outlier in test_splice_outliers:
+            outlier.update({
+                'chrom': '7', 'delta_psi': 0.85, 'gene_id': 'ENSG00000106554', 'rare_disease_samples_total': 20,
+                'rare_disease_samples_with_junction': 1, 'read_count': 1297, 'strand': '*', 'type': 'psi5', 'z_score': 12.34
+            })
+            RnaSeqSpliceOutlier(**outlier).save()
+        outlier_sample = Sample.objects.get(individual__guid=INDIVIDUAL_GUID, sample_type='RNA', tissue_type=None)
+        for gene_id in ['ENSG00000135954', 'ENSG00000135955', 'ENSG00000135956']:
+            outlier = {'sample': outlier_sample, 'gene_id': gene_id, 'z_score': 7.1, 'p_value': 0.0048, 'p_adjust': 0.001}
+            RnaSeqOutlier(**outlier).save()
```

Comment:
It creates new Sample and Outlier model objects to do the tests for the `isSignificant` tags set by Django `Window` and `DenseRank`. The samples usually are added to the fixture. But they seem to be too much for the fixture. So I add them here. Are there any better ways to do that?

Reply:
I think mocking the maximum number would help. So testing it with one value for the maximum number and then testing it again with a different number and seeing the difference. But also we probably still need to add ac ouple of fixtures, I think that would be better than adding data here

Reply:
I changed to add data to the fixtures.